### PR TITLE
Adding Grafana dashboard

### PR DESCRIPTION
**Pull request by tsurdilo**
Signed-off-by: Tihomir Surdilovic <tihomir@temporal.io>

Updated Prometheus and Grafana docker-compose configs to:
* enables server metrics
* set up automatic prometheus scraping point to temporal server metrics
* set up automatic grafana dashboard that can be used out of box
* updates the how-to doc to mention this

To test: run ./start then go to http://localhost:3000/ , log in as admin/admin and to go DashBoards->Home and click on the "Temporal" dashboard thats already there

### CODE REVIEW

**Comment by robholland** [comment]
Why build containers instead of using the upstream ones with volumes?

File: docker-compose.override.yml [hunk 1]
```diff
@@ -6,34 +6,30 @@ services:
...
   prometheus:
     container_name: lp-prometheus
-    image: prom/prometheus
+    build: './deployment/prometheus'
+    depends_on:
+      - temporal
     networks:
       - temporal-network
+    extra_hosts:
+      - "host.docker.internal:host-gateway"
```
Comment (robholland) [inline docker-compose.override.yml:21]: Why is this needed?

File: deployment/prometheus/config.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  scrape_interval: 5s
+  external_labels:
+    monitor: 'temporal-monitor'
+scrape_configs:
+  - job_name: 'prometheus'
+    metrics_path: /metrics
+    scheme: http
+    static_configs:
+      - targets:
+          - 'temporal:8000'
```
Comment (robholland) [inline deployment/prometheus/config.yml:11]: This is no longer scraping the worker.

File: docker-compose.yml [hunk 3]
```diff
@@ -39,11 +39,13 @@ services:
       - ENABLE_ES=true
       - ES_SEEDS=elasticsearch
       - ES_VERSION=v7
+      - PROMETHEUS_ENDPOINT=0.0.0.0:8000
```
Comment (robholland) [inline docker-compose.yml:42]: Please keep these changes in the override file. I'd prefer this docker-compose.yml be as close to upstream as possible. The only reason it's not an exact copy is that you cannot remove volume configuration with an override so I have to comment those lines out here.

File: docs/how-to-use.md [hunk 4]
```diff
@@ -25,6 +25,15 @@ cd background-checks
 
 This starts up a new Docker Application with a stack of containers.
 
+## How to monitor the application
+
+Once the applications has started you can navigate to [http://localhost:3000/](http://localhost:3000/)
+and log into Grafana with credentials `admin`, `admin`.
```
Comment (robholland) [inline docs/how-to-use.md:31]: I believe we can set grafana to not require any authentication, which is fine for our use case.

**Review by robholland — CHANGES_REQUESTED** [review]

File: deployment/prometheus/config.yml (hunk 2 above)
Comment (tsurdilo) [reply deployment/prometheus/config.yml:11]: what is the host:port worker is pushing to?

File: docker-compose.override.yml (hunk 1 above)
Comment (tsurdilo) [reply docker-compose.override.yml:21]: prob left over from testing will try without

File: docker-compose.yml (hunk 3 above)
Comment (tsurdilo) [reply docker-compose.yml:42]: hmmm we tell users to just add that as rly should be in main. will try in overrides, u got it :)

File: docs/how-to-use.md (hunk 4 above)
Comment (tsurdilo) [reply docs/how-to-use.md:31]: ok. i didnt know how will look

File: deployment/prometheus/config.yml (hunk 2 above)
Comment (robholland) [reply deployment/prometheus/config.yml:11]: Prometheus is pull, not push. This is the scrape target for the worker: https://github.com/temporalio/background-checks/pull/19/files#diff-66184e58408c7101ed064f2f630f09031b47b935716fdc2b496b90366f1a3b59L10 Note that it should be just `worker` now as I've removed the `lp` prefixes.

**Comment by tsurdilo** [comment]
@robholland 
managed to disable grafana login -- https://github.com/temporalio/background-checks/pull/19/commits/eb0754b88190d599cbf727827922b33a373ee29a

File: docker-compose.override.yml (hunk 1 above)
Comment (tsurdilo) [reply docker-compose.override.yml]: fixed

File: deployment/prometheus/config.yml (hunk 2 above)
Comment (tsurdilo) [reply deployment/prometheus/config.yml:11]: cant click on that link somehow. Just see - TEMPORAL_GRPC_ENDPOINT=lp-temporal:7233
not sure if thats what you are talking about

File: docs/how-to-use.md (hunk 4 above)
Comment (tsurdilo) [reply docs/how-to-use.md:31]: fixed, no more login screen

File: cli/bgc-backend/cmd/worker.go [hunk 5]
```diff
@@ -67,7 +67,7 @@ func newPrometheusScope(c prometheus.Configuration) tally.Scope {
 	scopeOpts := tally.ScopeOptions{
 		CachedReporter: reporter,
 		Separator:      prometheus.DefaultSeparator,
-		Prefix:         "background_checks",
+		Prefix:         "",
```
Comment (robholland) [inline cli/bgc-backend/cmd/worker.go:70]: This line can go as empty `Prefix` is the default.

File: deployment/prometheus/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,3 @@
+FROM prom/prometheus
```
Comment (robholland) [inline deployment/prometheus/Dockerfile:1]: No need to build a new prometheus image. Just mount your config into the container like the existing code does.

File: docker-compose.override.yml [hunk 7]
```diff
@@ -2,9 +2,14 @@ version: "3.5"
 services:
   temporal:
     environment:
-      - PROMETHEUS_ENDPOINT=0.0.0.0:9090
+      - PROMETHEUS_ENDPOINT=0.0.0.0:8000
+    ports:
+      - 8000:8000
```
Comment (robholland) [inline docker-compose.override.yml:7]: You don't need to expose the port, only prometheus is going to hit it and prometheus talks to the container directly.

File: docker-compose.override.yml [hunk 8]
```diff
@@ -42,6 +54,8 @@ services:
     restart: unless-stopped
     environment:
       - TEMPORAL_GRPC_ENDPOINT=temporal:7233
+    ports:
```
Comment (robholland) [inline docker-compose.override.yml:57]: Don't expose the port.

File: docs/how-to-use.md (hunk 4 above)
Comment (robholland) [reply docs/how-to-use.md:31]: Awesome, so please update the doc :)

File: deployment/prometheus/prometheus.yml [hunk 9]
```diff
@@ -7,4 +7,3 @@ scrape_configs:
     static_configs:
```
Comment (robholland) [inline deployment/prometheus/prometheus.yml:7]: You should replace this file with your `config.yml` and then you won't need to build a new container.

File: cli/bgc-backend/cmd/worker.go (hunk 5 above)
Comment (robholland) [reply cli/bgc-backend/cmd/worker.go:70]: Makes sense to remove the prefix, but lets make sure there is a label so we can tell these metrics are from the background check app.

**Review by robholland — CHANGES_REQUESTED** [review]
Looking good. Just a few tidy ups.
